client/blocks/comment-detail/index.jsx

```diff
@@ -82,16 +82,31 @@ export class CommentDetail extends Component {
 
 	toggleApprove = () => {
 		const { commentId, commentStatus, postId, setCommentStatus } = this.props;
-		setCommentStatus( commentId, postId, 'approved' === commentStatus ? 'unapproved' : 'approved' );
+		const isPersisted = 'approved' === commentStatus || 'unapproved' === commentStatus;
+
+		setCommentStatus( commentId, postId, 'approved' === commentStatus ? 'unapproved' : 'approved', {
```

Comment:
> why isn't siteId here?

It's passed directly in `mapDispatchToProps`.

> we can uniquely identify a single comment by siteId and commentId only

But the actions still require `postId`.